pcsx2/gui/Panels/BiosSelectorPanel.cpp

```diff
@@ -162,28 +167,49 @@ bool Panels::BiosSelectorPanel::ValidateEnumerationStatus()
 	return validated;
 }
 
-void Panels::BiosSelectorPanel::DoRefresh()
+void Panels::BiosSelectorPanel::EnumThread::ExecuteTaskInThread()
 {
-	if (!m_BiosList) return;
+	for (size_t i = 0; i < m_parent.m_BiosList->GetCount(); ++i)
+	{
+		wxString description;
+		if (!IsBIOS((*m_parent.m_BiosList)[i], description)) continue;
+		Result.emplace_back(std::move(description), i);
+	}
+
+	wxCommandEvent done(pxEvt_BiosEnumerationFinished);
+	done.SetClientData(this);
+	m_parent.GetEventHandler()->AddPendingEvent(done);
+}
 
+void Panels::BiosSelectorPanel::DoRefresh()
+{
 	m_ComboBox->Clear();
+	if (!m_BiosList->size()) return;
 
-	const wxFileName right(g_Conf->FullpathToBios());
-	bool biosSet = false;
+	m_ComboBox->Append(wxString("Enumerating BIOSes..."));
+	m_ComboBox->Update();
 
-	for(size_t i=0; i<m_BiosList->GetCount(); ++i)
-	{
-		wxString description;
-		if (!IsBIOS((*m_BiosList)[i], description)) continue;
-		int sel = m_ComboBox->Append( description, (void*)i );
+	m_EnumeratorThread.reset(new EnumThread(*this));
+
+	m_EnumeratorThread->Start();
+}
+
+void Panels::BiosSelectorPanel::OnEnumComplete(wxCommandEvent &evt)
+{
+	auto enumThread = static_cast<EnumThread*>(evt.GetClientData());
+	// Sanity check, in case m_BiosList was updated by ValidateEnumerationStatus() while the EnumThread was running
+	if (m_EnumeratorThread.get() != enumThread || m_BiosList->size() < enumThread->Result.size()) return;
+
+	const wxFileName& currentBios = g_Conf->FullpathToBios();
+	m_ComboBox->Clear(); // Clear the "Enumerating BIOSes..."
 
-		if (wxFileName((*m_BiosList)[i] ) == right)
-		{
+	for (const std::pair<wxString, u32>& result : enumThread->Result)
+	{
+		const int sel = m_ComboBox->Append(result.first, reinterpret_cast<void*>(result.second));
+		if (currentBios == wxFileName((*m_BiosList)[result.second]))
 			m_ComboBox->SetSelection(sel);
-			biosSet = true;
-		}
 	}
+};
 
-	if ((!biosSet) && !(m_ComboBox->IsEmpty()))
-		m_ComboBox->SetSelection(0);
-}
+Panels::BiosSelectorPanel::EnumThread::EnumThread(BiosSelectorPanel &parent) : m_parent(parent)
+{};
```

Comment:
Is this how our clang-format does it? that seems weird to me, might be because the function name is overly long though